tools/blocksconvert/builder/builder.go

```diff
@@ -38,10 +38,11 @@ type Config struct {
 	OutputDirectory string
 	Concurrency     int
 
-	ChunkCacheConfig cache.Config
-	UploadBlock      bool
-	DeleteLocalBlock bool
-	SeriesBatchSize  int
+	ChunkCacheConfig   cache.Config
+	UploadBlock        bool
+	DeleteLocalBlock   bool
+	SeriesBatchSize    int
+	TimestampTolerance int
```

Comment:
I would make it a `time.Duration`. Would be easier to understand how to use it.

Reply:
note that the `prometheus/tsdb` package itself is time-agnostic.

Reply:
Using `time.Duration` instead of `int` doesn't make it time specific. Maybe I'm not getting the feedback 🤔

Reply:
I think Julien's point is that TSDB is doesn't specify any time units. But Cortex (and Prometheus) only work with milliseconds  timestamps anyway. Here we're introducing this "knowledge" of timestamp units into blocks-builder.

Reply:
Sure, but for the Cortex final user I think it's easier to reason about time units than an `int`. You can convert `time.Duration` to milliseconds and keep the logic time-agnostic.

Reply:
> You can convert time.Duration to milliseconds and keep the logic time-agnostic.

It's not unit-agnostic if you convert it to milliseconds :) (Btw, I think `time.Duration` makes sense here)